src/display.cpp

```diff
@@ -233,7 +233,12 @@ display::display(const display_context * dc, std::weak_ptr<wb::manager> wb, repo
 
 	set_idle_anim_rate(preferences::idle_anim_rate());
 
-	zoom_index_ = std::distance(zoom_levels.begin(), std::find(zoom_levels.begin(), zoom_levels.end(), zoom_));
+	if(preferences::tile_size() > 0) {
+		zoom_ = preferences::tile_size();
+		validate_zoom_level_and_set_index(zoom_, zoom_index_);
```

Comment:
A comment in the new function says `This should never happen, since the value was already clamped earlier`, but that doesn't seem to be the case in this callsite. Furthermore, in the case that comment is reached, that function leaves the input parameter unmodified. Shouldn't this caller check the return value of the function, and if it is false, set `zoom_` to a default value?

Reply:
That's what I get for trying to minimise code changes and not properly understanding what is going on. XD However, looking at it now, if I set a default zoom and zoom_index in the false case, that will potentially change the behaviour of the `set_zoom()` calling function. I'd be inclined to just leave it as it is (and remove the comment that no longer applies with the refactoring).